sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java

```diff
@@ -307,64 +293,70 @@ public void update(int ordinal, Object value) {
 
     @Override
     public void setNullAt(int ordinal) {
```

Comment:
one question, does the rows returned by `ColumnarBatch.rowIterator` have to be mutable?